css-text-4/Overview.bs

```diff
@@ -5612,9 +5612,87 @@ Selecting How to Wrap: the 'text-wrap-style' property</h3>
 		<dt><dfn>pretty</dfn>
 		<dd>
 			Specifies the UA <em>should</em> bias for better layout over speed,
-			and is expected to consider multiple lines,
+			and is expected to consider multiple lines
 			when making break decisions.
-			Otherwise equivalent to ''text-wrap-style/auto'',
+			Otherwise equivalent to ''text-wrap-style/auto''.
+
+			The user agent may among other things
+			attempt to avoid excessively short last lines,
+			similarly to ''reduce-orphans'',
+			but it should also improve the layout in more ways.
+			The precise set of improvements is user agent dependent,
+			and may include things such as:
+			reducing the variation in length between lines;
+			avoiding <a href="https://en.wikipedia.org/wiki/River_(typography)">typographic rivers</a>;
+			prioritizing different classes of [=soft wrap opportunities=], [=hyphenation opportunities=], or [=justification opportunities=];
+			avoiding hyphenation on too many consecutive lines…
+
+			<div class=advisement>
+				The necessary computations may be expensive,
+				especially when applied to large amounts of text.
+				Authors are encouraged to asses the impact on performance
+				when using ''text-wrap-style: pretty'',
+				and possibly use it selectively where it matters most.
+			</div>
+
+		<dt><dfn>reduce-orphans</dfn>
```

Comment:
```suggestion
		<dt><dfn>avoid-orphans</dfn>
```

Reply:
cc: @astearns The key is to find a term that indicates that the value is about getting rid of orphans, but not in all circumstances, and not at any cost.
`avoid` as suggested here by @fantasai is already used in CSS (see https://drafts.csswg.org/indexes/) in a number of places with the meaning "don't, unless [X]", so I think it can work: the precise criteria for when not to do it is left up to the UA, but it is there.
To me, `reduce` could work too, but the meaning is almost too weak: instead of a rule generally applied with exceptions as implied by avoid, it would seem to indicate a slant, a preference. It's not wholly inappropriate here, especially given that everything is largely up to the user agent anyway, but that seems weaker than the author intent we're aiming for.

Reply:
> authors are going to expect that all orphans will be eliminated

If they don't think about too much, probably. But if you show them examples of cases where it would be bad to do so, and it doesn't do so, I suspect they'll be happy with the result, since forcibly doing it would be ugly.